.pylintrc

```diff
@@ -145,7 +145,9 @@ disable=invalid-name,
         deprecated-sys-function,
         exception-escape,
         comprehension-escape,
-        missing-module-docstring
+        missing-module-docstring,
+        too-many-ancestors,
```

Comment:
I think the `too-many-ancestors` warning is useful for new code and shouldn't be completely disabled.

Reply:
I disagree on that. Class ancestry is not some nit or small part of the code that can be overlooked (e.g. too-many-arguments) but it is one of the first things that will be checked by the class maintainer. Therefore, if this error is raised, it  is almost guaranteed that it is deliberate and with full knowledge of the implementer, and also extremely explicit to the PR reviewer (I doubt that this will go unnoticed in a PR). Therefore no need for a linter to enforce a rule on that, and litter all mixin classes with the disable-pylint statement.

Reply:
I still disagree on this, to be honest. I don't think that all of our current `too-many-ancestors` warnings are caused by deliberate choices that couldn't have been avoided.
Overlooking parts of the code is something that happens constantly when we're in a rush; using a linter is also a way to tell the developer to avoid coding in a hurry (or to add comments to notify readers when this is not possible). Disabling linter warnings on a module basis is ok for me, if I have to choose between the two situations.
So I personally still vote to keep this warning enabled. I'll leave it to the other reviewers anyway, as they might have different opinions.

Reply:
Still not persuaded by the comment. The only occurrences of too-many-ancestors that I have seen in the codebase are the Strategies / ExternalMixin (and I am also not sure about the Market classes, but I doubt that we have reached the limit yet), which is a design decision to implement it this way, therefore will not change. If you find another such occurrence then we can talk more about it, because personally I have not seen any.

Reply:
Warnings that pop up only in cases that we know are OK (which IMO is one of them, as I said this warning is only in one place in the codebase which we know is ok, and if new code is added that violates this we will definitely notice in the PR and make sure that it makes sense or not) are not really helpful, and will only end up polluting our codebase with pylint: disable with no benefit at all. 

In addition to having no benefit of leaving it there, this is contrary to the main goal of Pylint, which is to tailor it to your codebase's needs (that's why it is feasible to disable its warnings in so many ways after all). Essentially Pylint's goal is to not treat all its warnings as errors, but to explicitly select which ones should be treated as errors and which should not according to the codebase. 

BTW this comment is not only specific to this warning. IMO it probably makes sense to disable other warning types too, either because we disagree with them, or because they do not fit to our codebase (for instance, one such candidate is too-broad-exception, but I will leave this discussion for another comment :D )